### PR TITLE
Eventhub consumer throughput analysis

### PR DESCRIPTION
**Pull request by mahluk**
Includes results and graphs that explain the throughput analysis of event hub consumer(Event Processor Host Library) for different scenarios.

### CODE REVIEW

**Review by savas-guven — APPROVED** [review]

File: _posts/2019-12-24-eventhub-consumer-throughput-analysis.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
+---
+layout: post
+title: Event Hub Consumer Throughput Analysis 
+author: haluk.aktas@deepnetwork.om
```
Comment (yufufi) [inline _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:4]: @mahluk , you need to define yourself as an author in https://github.com/deepnetworkgmbh/blog/blob/master/_data/authors.yml and refer to that here.

File: _posts/2019-12-24-eventhub-consumer-throughput-analysis.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+---
+
+## Introduction
```
Comment (yufufi) [inline _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:7]: This wouldn't work on the main page. Did you try running the blog locally before sending the PR? There is already a title to the blog post (specified in `title`) and then the beginning is the `introduction`.

File: _posts/2019-12-24-eventhub-consumer-throughput-analysis.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+
+The average total duration is 30.69 msec/event and it is approximately 0.031 sec/event. The irate graph average value is 31.95 event/sec. When we compare these two, we can interpret 31.95 event/sec as 0.032 sec/event. Since results are not exact by the nature of Prometheus, we can see 0.032 sec/event for our computation and 0.031 sec/event for irate correlates. Our purpose to demonstrate this is that from this point on, we can rely on our irate graph to measure throughput after each improvement to the event hub consumer code.
+
+
+### 1.2	The Message Processing Loop 
+
+The related code part for getting the above results is in the following:
+
+```
```
Comment (yufufi) [inline _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:56]: add `c#` here to syntax highlighting for your inline code

image

image

Comment (yufufi) [reply _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:56]: Please do the same for other snippets.

File: _posts/2019-12-24-eventhub-consumer-throughput-analysis.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+
+### 2.1	Changes in Metrics
+
+Same metrics are used.
+
+### 2.1.1 Irate
+
+As we can see in the following irate graph, the rate of increase is increased when we compare it with the baseline scenario. The average rate value for the graph in baseline was 31.95 event/sec but the current graph displays average as 43.95 event/sec.
+
+![inloop parallel irate.](/blog/images/eventhub-consumer-blog-ss/inloop_parallel_irate.png)
```
Comment (yufufi) [inline _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:109]: Casing of your file names are not correct. `PNG` vs `png`. Check against the actual image files you've checked in.

File: _posts/2019-12-24-eventhub-consumer-throughput-analysis.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+
+### 4.3 Comparison Results Starting with Baseline Scenario
+
+|      Metric Name         |    Unit    |	Baseline (Avg)	| Inloop Parallel (Avg)	|	Outloop Parallel (Avg)	|	In & Out Loop Parallel (Avg)	|
+|        ---        	   |     ---    |		:---:		|			:---:		|			:---:			|				:---:				|
+| total_stopwatch          | msec/event |		`30.69`		|			`21.14`		|			`9`				|				`8`					|
+| servicebus_stopwatch     | msec/event |		`18.52`		|			`17.87`		|			`45`			|				`39`				|
+| blobstorage_stopwatch    | msec/event |		`9.35`		|			`9.31`		|			`20`			|				`13`				|
+| checkpoint_stopwatch 	   | msec/event |		`2.47`		|			`2.47`		|			`2`				|				`2`					|
+| processed_messagecounter | event/sec  |		`31.95`		|			`43.93`		|			`115.89`		|				`120`				|
```
Comment (yufufi) [inline _posts/2019-12-24-eventhub-consumer-throughput-analysis.md:367]: Could you please add a conclusion to your post?

**Review by yufufi — COMMENTED** [review]
Please start a new PR as this is already merged.

**Pull request merged**